cdap-common/src/main/resources/cdap-default.xml

```diff
@@ -1587,4 +1587,52 @@
     </description>
   </property>
 
+  <!-- Dataset extensions -->
+  <property>
+    <name>dataset.extensions.dir</name>
+    <value>/opt/cdap/ext/lib</value>
+  </property>
+
+  <property>
+    <name>dataset.extensions.modules</name>
+    <value>name1:classname1,name2:classname2,...</value>
```

Comment:
Since by default there wouldn't be any extension modules, I would say leave this property out and just have it in documentation (see my next comment below).